pkg/clients/iam/iam_client_test.go

```diff
@@ -248,4 +248,202 @@ var _ = Describe("IAMClient", func() {
 			Expect(client).To(BeNil())
 		})
 	})
+
+	Describe("RevokeBucketAccess", func() {
+		var mockIAM *mock.MockIAMClient
+
+		BeforeEach(func() {
+			mockIAM = &mock.MockIAMClient{}
+		})
+
+		It("should fail on non-existent user gracefully", func(ctx SpecContext) {
```

Comment:
The test says that it should fail "gracefully" but doesn't seem to check the gracefulness of the error (i.e. it tests the same way than the next test).

Reply:
Thank you for your feedback — it helped me catch a bug. I’ve added the change along with tests: if an access key is deleted between listing and deletion, we now handle it by returning a “not found” error and continuing with the list.

Additionally, I’ve created a ticket for next week to review the graceful handling of all errors, and I’ve added it to the next sprint: [S3C-9439](https://scality.atlassian.net/browse/S3C-9439).